apps/backend/src/modules/users/user.service.ts

```diff
@@ -88,7 +104,77 @@ class UserService implements Service {
 	): Promise<null | UserDto> {
 		const user = await this.userRepository.update(id, payload);
 
-		return user ? user.toObject() : null;
+		return user?.toObject() ?? null;
+	}
+
+	public async updateAvatar(
+		userId: number,
+		avatarFile?: File,
+	): Promise<null | UserDto> {
+		if (!avatarFile) {
+			throw new UserError({
+				message: ErrorMessage.FILE_MISSING,
+				status: HTTPCode.BAD_REQUEST,
+			});
+		}
+
+		const user = await this.userRepository.find(userId);
+
+		if (!user) {
+			throw new UserError({
+				message: ErrorMessage.UNAUTHORIZED,
+				status: HTTPCode.UNAUTHORIZED,
+			});
+		}
```

Comment:
We will always have a user at this point. Otherwise authorization plugin will throw an error